bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/FileInfo.java

```diff
@@ -102,14 +101,34 @@ synchronized Long getLastAddConfirmed() {
         return lac;
     }
 
-    synchronized long setLastAddConfirmed(long lac) {
-        if (null == this.lac || this.lac < lac) {
-            this.lac = lac;
+    long setLastAddConfirmed(long lac) {
+        long lacToReturn;
+        synchronized (this) {
+            if (null == this.lac || this.lac < lac) {
+                this.lac = lac;
+                setChanged();
+            }
+            lacToReturn = this.lac;
+        }
+        LOG.trace("Updating LAC {} , {}", lacToReturn, lac);
+
+
+        notifyObservers(new LastAddConfirmedUpdateNotification(lacToReturn));
```

Comment:
Would it be easy to avoid allocating the wrapper object `LastAddConfirmedUpdateNotification`, at least in the case where there are no registered observers?